File: library/src/main/java/com/silverhetch/aura/view/fab/FabControl.java
package com.silverhetch.aura.view.fab;

/**
 * Object that controls a floating action button on screen.
 */
public interface FabControl {
    /**
     * Attach a new behavior with new [FabBehavior]
     */
    void attachFab(FabBehavior fabBehavior);

    /**
     * Detach from fab.
     */
    void detachFab(FabBehavior fabBehavior);
}

File: library/src/main/java/com/silverhetch/aura/view/fab/FabControl.kt
package com.silverhetch.aura.view.fab

/**
 * Object that controls a floating action button on screen.
 */
interface FabControl {
    /**
     * Attach a new behavior with new [FabBehavior]
     */
    fun attachFab(fabBehavior: FabBehavior)

    /**
     * Detach from fab.
     */
    fun detachFab()
}
